### ISSUE TITLE
Connections Usage

### ISSUE THREAD

**Issue by Jonatthu** [graphql-dotnet/graphql-dotnet#352]
This is the only way that I can make connection field work
```

            Connection<ClientPlaylistType>()
                .Name("PlaylistConnection")
                .Resolve(x =>
                {
                    var list = new List<ClientPlaylist>
                    {
                        new ClientPlaylist { Id = 3 }
                    };
                    return new {
                        Items = list,
                        Edges = list, // TODO: node, cursor,
                        TotalCount = 5
                    };
                });
```

What I was reading from here and the samples are not working or are outdated
https://github.com/graphql-dotnet/graphql-dotnet/issues/40

How is the right way to use this?

**Comment by MFFoX** [comment]
+1 I'd love some general usage of Connections as well.

**Comment by joemcbride** [comment]
I don't know Relay myself.  I would suggest to have a look at https://github.com/graphql-dotnet/relay

**Comment by MFFoX** [comment]
Thanks, Joe. I am not actually using Relay, but just want to implement cursor-based pagination. I _think_ the `ConnectionBuilder` and basic support for connections in this repository will work for my use case, but I can't seem to figure out how to resolve the object correctly.

**Comment by MFFoX** [comment]
I will try installing and using the relay library for now. If those classes aren't going to supported/documented, maybe they should be completely removed in favor of graphql-dotnet/relay?

**Comment by joemcbride** [comment]
I have certainly had the same thought.  They were originally added by contributors.  Here are some tests that may help with usage:

https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL.Tests/Builders/ConnectionBuilderTests.cs

**Comment by joemcbride** [comment]
I am also vaguely recalling that someone had to add `ConnectionType<>` and `EdgeType<>` to their DI container for types to be resolved through DI.

**Comment by joemcbride** [comment]
See #451

**Comment by SimonCropp** [comment]
think this was fixed as part of #451 so closing.

**Comment by sungam3r** [comment]
@SimonCropp  By the way about #451 . I stumbled upon it a while ago and thought about whether it is worth leaving it where it is now or to remove it at all. For such things there is already this repository - https://github.com/graphql-dotnet/relay . Where does the area of responsibility lie? I tend to remove all references to Relay specific types from this repo. What do you think?
